src/vcUnitConversion.cpp

```diff
@@ -0,0 +1,174 @@
+  return sourceValue * scalar;
+}
+
+double vcUnitConversion_ConvertVolume(double sourceValue, vcVolumeUnit sourceUnit, vcVolumeUnit requiredUnit)
+{
+  double scalar = 1.0;
+
+  if (sourceUnit == vcVolumeUnit_CubicMeter)
+    scalar = 1.0;
+  else if (sourceUnit == vcVolumeUnit_USSGallons)
+    scalar = 1 / 264.12;
+  else if (sourceUnit == vcVolumeUnit_USSQuart)
+    scalar = 1 / 1056.68;
+  else if (sourceUnit == vcVolumeUnit_Liter)
+    scalar = 1 / 1000;
+  else if (sourceUnit == vcVolumeUnit_CubicInch)
+    scalar = 1.0 / 61023.74;
+  else if (sourceUnit == vcVolumeUnit_CubicFoot)
+    scalar = 1 / 35.31;
+  else if (sourceUnit == vcVolumeUnit_MegaLiter)
+    scalar = 1000;
+  else if (sourceValue == vcVolumeUnit_CubicYard)
+    scalar = 1 / 1.3;
+
+  if (requiredUnit == vcVolumeUnit_CubicMeter)
+    scalar = 1.0;
+  else if (requiredUnit == vcVolumeUnit_USSGallons)
+    scalar = 264.12;
+  else if (requiredUnit == vcVolumeUnit_USSQuart)
+    scalar = 1056.68;
+  else if (requiredUnit == vcVolumeUnit_Liter)
+    scalar = 1000;
+  else if (requiredUnit == vcVolumeUnit_CubicInch)
+    scalar = 61023.74;
+  else if (requiredUnit == vcVolumeUnit_CubicFoot)
+    scalar = 35.31;
+  else if (requiredUnit == vcVolumeUnit_MegaLiter)
+    scalar = 1 / 1000;
+  else if (requiredUnit == vcVolumeUnit_CubicYard)
+    scalar = 1 / 1.3;
+
+  return sourceValue;
+}
+
+double vcUnitConversion_ConvertSpeed(double sourceValue, vcSpeedUnit sourceUnit, vcSpeedUnit requiredUnit)
+{
+
+double scalar = 1.0;
+
+  if (sourceUnit == vcSpeed_MetresPerSecond)
+    scalar = 1.0;
+  else if (sourceUnit == vcSpeed_KilometresPerHour)
+    scalar = 1.0 / 3.6;
+  else if (sourceUnit == vcSpeed_USSurveyMilesPerHour)
+    scalar = 1.0 / 2.2;
+  else if (sourceUnit == vcSpeed_NauticalMilesPerHour)
+    scalar = 1.0 / 1.9;
+  else if (sourceUnit == vcSpeed_FootPerSecound)
+    scalar = 1.0 / 3.2;
+
+  // There are more units here
+
+  if (requiredUnit == vcSpeed_MetresPerSecond)
+    scalar = 1.0;
+  else if (requiredUnit == vcSpeed_KilometresPerHour)
+    scalar = 3.6;
+  else if (requiredUnit == vcSpeed_USSurveyMilesPerHour)
+    scalar = 2.2;
+  else if (requiredUnit == vcSpeed_NauticalMilesPerHour)
+    scalar = 1.9;
+  else if (requiredUnit == vcSpeed_FootPerSecound)
+    scalar = 3.2;
+  return sourceValue * scalar;
+}
+
+double vcUnitConversion_ConvertTemperature(double sourceValue, vcTemperatureUnit sourceUnit, vcTemperatureUnit requiredUnit)
+{
+  double scalar = 1.0;
+
+  if (sourceUnit == vcTemperature_Celcius)
+    scalar = 1.00;
+  else if (sourceUnit == vcTemperature_Kelvin)
+    scalar = sourceValue - 273.15;
+  else if (sourceUnit == vcTemperature_Farenheit)
+    scalar = (sourceValue - 32) * (5 / 9);
+  if (requiredUnit == vcTemperature_Celcius)
+    scalar = 1.00;
+  else if (requiredUnit == vcTemperature_Kelvin)
+    scalar = sourceValue + 273.15;
+  else if (requiredUnit == vcTemperature_Farenheit)
+    scalar = (sourceValue * (5 / 9)) + 32;
```

Comment:
This line has an issue.  In strongly typed languages (c++), `5` (integer) and `5.0` (double) are different types. 
`5 / 9` will truncate to `0`, because integers are whole numbers. Whereas `5.0 / 9.0` will equal `0.5555556`. 

This is a simple fix, if you simply change the line to only have doubles: `scalar = (sourceValue * (5.0 / 9.0)) + 32.0;`